cmd/dagger/query.go

```diff
@@ -89,42 +89,18 @@ func Query(cmd *cobra.Command, args []string) {
 }
 
 func doQuery(ctx context.Context, query, op string, vars map[string]interface{}) ([]byte, error) {
-	opts := []dagger.ClientOpt{
-		dagger.WithWorkdir(workdir),
-		dagger.WithConfigPath(configPath),
-	}
-
-	if debugLogs {
-		opts = append(opts, dagger.WithLogOutput(os.Stderr))
-	}
-	c, err := dagger.Connect(ctx, opts...)
-	if err != nil {
-		return nil, err
-	}
-	defer c.Close()
-
 	res := make(map[string]interface{})
-	resp := &dagger.Response{Data: &res}
-	err = c.Do(ctx,
-		&dagger.Request{
-			Query:     query,
-			Variables: vars,
-			OpName:    op,
-		},
-		resp,
-	)
+	err := withEngine(ctx, "", func(ctx context.Context, r *router.Router) error {
+		_, err := r.Do(ctx, query, op, vars, &res)
```

Comment:
General note: we should keep an eye on `router.Do`, it's been used only internally so far, I don't 100% know the error reporting etc are on par with regular requests